src/components/Results.jsx

```diff
@@ -1,53 +1,93 @@
 import React from "react";
+import { useDispatch, useSelector } from "react-redux";
 import Container from "./utils/Container";
 import TextBox from "./utils/TextBox";
 import Icon from "./utils/Icon";
 import Button from "./utils/Button";
-import CssVariables from "./utils/cssVariables";
+import { CPU, getMarkColor } from "../utils/mixin";
+import {
+  gameMode,
+  playerMark,
+  startNewGame,
+  winnerMark,
+} from "../reducers/gameSlice";
+import langs from "../langs/langs";
 
 const Results = (props) => {
-  const winner = "x"; //test cases
+  const dispatch = useDispatch();
+  const winner = useSelector(winnerMark);
+  const player1Mark = useSelector(playerMark);
+  const mode = useSelector(gameMode);
+  const markColor = winner === "x" ? getMarkColor("x") : getMarkColor("o");
 
-  const color = winner === "x" ? CssVariables.blue : CssVariables.orange;
-  const textColor = winner === "x" ? "text-blue" : "text-orange";
+  let textColor = "text-silver";
+  if (winner) {
+    textColor = winner === "x" ? "text-blue" : "text-orange";
+  }
+
+  const quitGameHandler = () => {
+    dispatch(startNewGame({ isPlaying: false, resetScores: true }));
+  };
+
+  const nextRoundHandler = () => {
+    dispatch(startNewGame({ isPlaying: true, resetScores: false, mode: mode }));
+  };
+
+  let playerInfoText = "";

```

Comment:
You are trying to do too much at once. Single merge request should aim to achieve one thing at once. Working on multiple things and starting next one before finishing previous one only slows your progress. Please work in organised manner, it will help for sure

Here are some simple, custom implementation ideas for language change:

https://wavelop.com/en/story/implementing-multi-language-without-any-library-in-react-hooks/
https://betterprogramming.pub/react-context-hooks-part-2-ui-language-switch-f4610a21379b